Automattic/vip-go-mu-plugins
# Add 'vip data-cleanup' CLI commands

## Description

Introduces a set of new CLI commands, to be used programmatically after the relevant jobs are run:
- `wp vip data-cleanup sql-import`
- `wp vip data-cleanup datasync`

The functionality seen here largely comes from [vip-migrations.php](https://github.com/Automattic/vip-go-mu-plugins/blob/master/vip-helpers/vip-migrations.php). Currently after a data-sync we run the `vip_after_data_migration` hook (via cron). After a SQL import, we run [vip migration cleanup](https://github.com/Automattic/vip-go-mu-plugins/blob/master/wp-cli/vip-migrations.php#L22). This changeset is a move towards standardizing what we are doing with specific intentions after these jobs, with the motivation of removing the legacy "migration" functionality that came during the era of WPcom -> VIP Go migrations. The removal can start to take place after all sites are off of VIPd.

### Relevant Changes

Of note, nothing will immediately change until these CLIs are implemented on the jobs-side. With that said, here are the changes of note:

1. After SQL imports, we are no longer [deleting certain options](https://github.com/Automattic/vip-go-mu-plugins/blob/master/wp-cli/vip-migrations.php#L66-L79). This is instead handled on the jobs side of things.
1.  After SQL imports and during single-site datasyncs, we now run `dbDelta( 'global' )`. This is pretty important, as currently we miss checks against the user/usermeta tables on single sites, and checks against multisite-specific tables after SQL imports.
1. The documented `vip_go_migration_cleanup` hook is no longer run after SQL imports (this was unintended).
1. The `vip_after_data_migration` hook is no longer needed. This one was unintended to be used by customers, but is in use by one customer currently. The use-case is valid, but we'll be solving their need in another way and can reach out directly to resolve the backwards-compatibility change there.

### Proposed Changes

Would like to also introduce new hooks that are more specific to the action taking place (deprecating vip_go_migration_cleanup). These will come with documentation updates (naming feedback welcome 🙂 ):

- `vip_datasync_cleanup`
- `vip_sqlimport_cleanup`

## Changelog Description

### Data cleanup: SQL Imports & Data Syncs

Introducing new CLI commands and hooks that will be used after data syncs and after SQL imports.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
1. Run `wp vip data-cleanup datasync`
1. Run `wp vip data-cleanup sql-import`
